src/extensions/default/HealthData/HealthDataManager.js

```diff
@@ -0,0 +1,170 @@
+
+        if (!userUuid) {
+            userUuid = uuid.v4();
+            PreferencesManager.setViewState("UUID", userUuid);
+        }
+
+        oneTimeHealthData.uuid = userUuid;
+        oneTimeHealthData.snapshotTime = (new Date()).getTime();
+        oneTimeHealthData.os = brackets.platform;
+        oneTimeHealthData.userAgent = navigator.userAgent;
+        oneTimeHealthData.osLanguage = brackets.app.language;
+        oneTimeHealthData.bracketsLanguage = brackets.getLocale();
+        oneTimeHealthData.bracketsVersion = brackets.metadata.version;
+
+        HealthDataUtils.getInstalledExtensions()
+            .done(function (userInstalledExtensions) {
+                oneTimeHealthData.installedExtensions = userInstalledExtensions;
+            })
+            .always(function () {
+                return result.resolve(oneTimeHealthData);
+            });
+
+        return result.promise();
+    }
+
+    /**
+     * Send data to the server
+     */
+    function sendHealthDataToServer() {
+        var result = new $.Deferred();
+        
+        getHealthData().done(function (healthData) {
+            
+            var url = brackets.config.healthDataServerURL,
+                data = JSON.stringify(healthData);
+
+            $.ajax({
+                url: url,
+                type: "POST",
+                data: data,
+                dataType: "text",
+                contentType: "text/plain"
+            })
+                .done(function () {
+                    result.resolve();
+                })
+                .fail(function (jqXHR, status, errorThrown) {
+                    console.error("Error in sending health data. Response : " + jqXHR.responseText + ". Status : " + status + ". Error : " + errorThrown);
+                    result.reject();
+                });
+        })
+            .fail(function () {
+                result.reject();
+            });
+
+        return result.promise();
+    }
+
+    /*
+     * Check if the health data is to be sent to the server. If the user has enabled tracking, health data will be sent once every 24 hours.
+     * Send health data to the server if the period is more than 24 hours.
+     * We are sending the data as soon as the user launches brackets. The data will be sent to the server only after the notification dialog 
+     * for opt-out/in is closed.
+     */
+    function checkHealthDataSend() {
+        var result = new $.Deferred(),
+            isHDTracking = prefs.get("healthDataTracking"),
+            notificationDialogShown = PreferencesManager.getViewState("healthDataNotificationShown");
+        
+        window.clearTimeout(timeoutVar);
+        if (isHDTracking && notificationDialogShown) {
+            var lastTimeSent = PreferencesManager.getViewState("lastTimeSentData"),
+                currentTime = (new Date()).getTime();
+
+            if (!lastTimeSent || (currentTime >= lastTimeSent + ONE_DAY)) {
```

Comment:
This logic changed a lot since the original review, and there's a big behavior difference: on first launch, `"lastTimeSentData"` isn't set so we now go into this case and send data _immediately_.  My understanding was we wanted to wait 24 hours before sending the first ping -- both so the user has more time to opt out (which is _critical_ if we decide to change the initial notification UI), and also so the first data ping actually contains useful data (once it contains info like average project size, etc.).

The safest way to do that is to immediately set `"lastTimeSentData"` to the current timestamp (`Date.now()`) if we get here and it's not set yet.  That's similar to what the original code did, before commit 07320528.

If it feels too weird to initially set the pref _as if_ data was sent when in fact we haven't sent any yet, you could change the semantics so the pref becomes `"nextHealthDataSendTime"` -- which I think could actually make this code a little easier to understand anyway.

This code is pretty delicate, so be careful if you decide to fix this a different way: I think I see some pitfalls here where other bugs could easily be introduced (like doing math on undefined, or never sending any data if the app isn't left open for > 24h nonstop).